### ISSUE TITLE
Nuget restore does not update packages.lock.json when uninstalling a package

### ISSUE THREAD

**Issue by amybingzhao** [NuGet/Home#8091]
## Details about Problem

NuGet product used: NuGet.exe, VS UI
NuGet version (x.x.x.xxx): 4.9.4.5839
MSBuild version: 15.9.21.664
VS version (if appropriate): VS 2017 (Professional Edition)
OS version: windows 7

## Detailed repro steps so we can see the same problem

1. In a project using the dotnet core SDK csproj, confirm that a packages.lock.json file exists and that the csproj has <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>.

2. Remove a PackageReference by deleting it from the csproj OR by uninstalling through the VS Package Manager GUI.

3. Rebuild

4. Run `nuget restore <solutionPath> -Verbosity detailed` from the command line -->output says "The restore inputs for 'ProjectName' have not changed. No further actions are required to complete the restore. Committing restore...
Assets file has not changed. Skipping assets file writing. Path: Path\to\project\proj                                                                                                                         ect.assets.json
No-Op restore. The cache will not be updated." 

Expected behavior: a nuget restore is performed (not a no-op restore) and the packages.lock.json is updated to remove the uninstalled package as a dependency.

**Comment by nkolev92** [comment]
Thanks for creating this issue. 

This is a duplicate of https://github.com/NuGet/Home/issues/7930, which we have fixed in 5.1 release, which is matching the 16.1 release of Visual Studio. 

A quick explanation about what you are seeing. 

> 2. Remove a PackageReference by deleting it from the csproj OR by uninstalling through the VS Package Manager GUI.

At this point, after the bug fix, it will update the lock file. 

> 3. Rebuild

Since you are using a lock file and you have no changed anything in your project, nothing should be updated. 

> 4. Run nuget restore <solutionPath> -Verbosity detailed from the command line -->output says "The restore inputs for 'ProjectName' have not changed. No further actions are required to complete the restore. Committing restore...
Assets file has not changed. Skipping assets file writing. Path: Path\to\project\proj ect.assets.json
No-Op restore. The cache will not be updated."

This is actually to be expected. Restore should have already run and regenerated the lock file in step 2. 

Basically 2. is the one where you are hitting the issue.
